src/regl.js

```diff
@@ -36,11 +39,38 @@ const Regl = ({ style, aspect, children }) => {
     resize.current()
 
     if (!regl.current) {
-      regl.current = _regl({
-        container: container.current,
-        extensions: ['OES_texture_float', 'OES_element_index_uint'],
-      })
-      setReady(true)
+      try {
+        const canvas = document.createElement('canvas')
+        const context =
+          canvas.getContext('webgl') || canvas.getContext('experimental-webgl')
+        if (!context) {
+          throw new Error('WebGL is not supported in this browser.')
+        }
+
+        const requiredExtensions = [
+          'OES_texture_float',
+          'OES_element_index_uint',
+        ]
+        const missingExtensions = requiredExtensions.filter(
+          (ext) => !context.getExtension(ext)
+        )
+
+        if (missingExtensions.length > 0) {
+          console.error(`missing extensions: ${missingExtensions.join(', ')}`)
+          throw new Error(
+            `Required WebGL extensions not supported: ${missingExtensions.join(
+              ', '
+            )}`
+          )
+        }
+        regl.current = _regl({
+          container: container.current,
+          extensions: requiredExtensions,
+        })
+        setReady(true)
+      } catch (error) {
+        setError("Your device doesn't support this figure")
```

Comment:
nit: maybe we use slightly more generic language?

```suggestion
        setError("Your device doesn't support rendering this map.")
```